### PR TITLE
[usage] remove unused GetUpcomingInvoice

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Removes no longer needed `getUpcomingInvoice` API.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Review by easyCZ — COMMENTED** [review]
There are existing calls to this service, I believe we can only remove once there are no more calls to this service (and in different chagesets).

Screenshot 2022-09-19 at 8 31 47

[Grafana](https://grafana.gitpod.io/d/fDH4_QMVk/grpc-server?orgId=1&var-service=usage.v1.BillingService&var-method=GetUpcomingInvoice&var-type=All)

In an incident on Friday September 16, 2022, we've had to patch version of server. Had we removed the implementation of `SetBilledInstance` in the same changeset, server would now fail on starting workspaces. The same applies here, we need to remove all usage before we land the removal.

**Comment by svenefftinge** [comment]
I have removed the call site in https://github.com/gitpod-io/gitpod/pull/12997 on Friday. 
So once we have deployed that, we should be able to land this one.

/hold

**Review by easyCZ — APPROVED** [review]
There are no [more API calls ](https://grafana.gitpod.io/d/fDH4_QMVk/grpc-server?orgId=1&var-service=usage.v1.BillingService&var-method=GetUpcomingInvoice&var-type=All&from=now-30m&to=now)after the last rollout.

**Pull request merged**
